src/main/java/producer/ProducerRecord.java

```diff
@@ -0,0 +1,156 @@
+/*
+A key/value pair to be sent to Kafka.
+This consists of a topic name to which the record is being sent, an optional partition number, and an optional key and value.
+
+If a valid partition number is specified that partition will be used when sending the record.
+
+If no partition is specified but a key is present a partition will be chosen using a hash of the key. If neither key nor partition is present a partition will be assigned in a round-robin fashion.
+
+The record also has an associated timestamp. If the user did not provide a timestamp, the producer will stamp the record with its current time. The timestamp eventually used by Kafka depends on the timestamp type configured for the topic.
+ */
+package producer;
+import commons.header.Header;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.*;
+
+public class ProducerRecord<K, V> {
+    private String topic;
+    private Long timestamp;
+    private V value;
+    private Optional<Integer> partitionNumber = Optional.empty();
+    private Optional<K> key = Optional.empty();
+    private Iterable<Header> headers;
+
+    public ProducerRecord(String topic, Integer partition, Long timestamp, K key, V value) {
+        this.topic = topic;
+        this.timestamp = timestamp;
+        this.value = value;
+        this.partitionNumber = Optional.ofNullable(partition);
+        this.key = Optional.ofNullable(key);
+    }
+
+    public ProducerRecord(String topic, Integer partition, Long timestamp, K key, V value, Iterable<Header> headers) {
```

Comment:
what does the `timestamp` represent?

Reply:
I made this ticket as close as to the [Apache Kafka docs ](https://kafka.apache.org/25/javadoc/org/apache/kafka/clients/producer/ProducerRecord.html)on ProducerRecord, I think since its a entry ticket to the class, its fine if we leave it, worse case, we can remove it afterwards once we circle back to this.

### Screenshot 
<img width="1157" alt="Screenshot 2024-12-27 at 8 48 07 PM" src="https://github.com/user-attachments/assets/ec02af30-48f3-42b6-af72-54d30d93e2c1" />

Reply:
ideally each record will have its own auto-generated `timestamp` when it gets created, not rlly sure on my end how much the real kafka utilizes the constructors with the timestamp param but it can be useful for testing purposes

Reply:
Got it